src/Symfony/Component/Cache/Traits/PdoTrait.php

```diff
@@ -395,6 +396,7 @@ private function getConnection()
             } else {
                 switch ($this->driver = $this->conn->getDriver()->getName()) {
                     case 'mysqli':
+                        throw new NotSupportedException(sprintf('The adapter "%s" does not support the mysqli driver, use pdo_mysql instead.', \get_class($this)));
```

Comment:
we're going to use a base `\LogicException` directly, no need for the specialized one (especially as a bug fix)